### ISSUE TITLE
Unable to compile due to Smithay source/patch

### ISSUE THREAD

**Issue by kevinbburns** [pop-os/cosmic-epoch#82]
After reading the instructions and trying to compile, I am getting this error:

    Updating git repository `https://github.com/smithay//smithay`
error: failed to load source for dependency `smithay`

Caused by:
  Unable to update https://github.com/smithay//smithay?rev=09e29418e5#09e29418

Using cargo, I have compiled every single repo and even the smithay project without issue.

Does anyone else experience and find a workaround?

**Comment by User8395** [comment]
This error appeared in the CI too

**Comment by kschurter** [comment]
The first issue is that the commit hash in the smithay repo that is referenced in `cosmic-comp/Cargo.toml` is not automatically fetched because it does not belong to any branches. Cargo would did not successfully implement the patch. So I tried using a local clone of the repo after forcing git to fetch the commit

```
git clone https://github.com/Smithay/smithay
cd smithay
git fetch 09e294
git reset --hard 09e294
```
I tried compiling cosmic-epoch again after changing the patch in `cosmic-comp/Cargo.toml` to `smithay = { path = PATH_TO_SMITHAY}`. Cargo stopped complaining about a bad revspec, but the smithay project did not compile in that state.

I found that changing the branch in the cosmic-comp repo does allow the install to complete:

```
git clone --recurse-submodules https://github.com/pop-os/cosmic-epoch
cd cosmic-epoch/cosmic-comp
git checkout main_jammy
cd ..
just sysext
```
